packages/botonic-cli/src/analytics/credentials-handler.ts

```diff
@@ -0,0 +1,123 @@
+import { homedir } from 'os'
+import { join, resolve } from 'path'
+import { v4 as uuidv4 } from 'uuid'
+
+import {
+  BOT_CREDS_FILENAME,
+  BOTONIC_HOME_DIRNAME,
+  BOTONIC_PROJECT_PATH,
+  GLOBAL_CREDS_FILENAME,
+} from '../constants'
+import {
+  BotCredentials,
+  CredentialsHandlerArgs,
+  GlobalCredentials,
+} from '../interfaces'
+import { create, pathExists, readJSON, writeJSON } from '../util/file-system'
+import { execCommand, isWindows } from '../util/processes'
+
+export class CredentialsHandler {
+  homePath: string
+  pathToCredentials: string
+
+  constructor(args: CredentialsHandlerArgs) {
+    this.homePath = resolve(args.homePath)
+    this.pathToCredentials = join(this.homePath, args.filename)
+    this.initialize()
+  }
+
+  initialize(): void {
+    this.createIfNotExists()
+  }
+
+  generateId(): string {
+    return uuidv4()
+  }
+
+  createIfNotExists(): void {
+    if (!pathExists(this.homePath)) create(this.homePath)
+  }
+
+  load(): any {
+    try {
+      return (
+        pathExists(this.pathToCredentials) && readJSON(this.pathToCredentials)
+      )
+    } catch (e) {
+      console.warn('Credentials could not be loaded')
+      return undefined
+    }
+  }
+
+  dump(obj: any): void {
+    try {
+      writeJSON(this.pathToCredentials, obj)
+    } catch (e) {
+      console.warn('Credentials could not be overwritten')
+    }
+  }
+}
+
+export class GlobalCredentialsHandler extends CredentialsHandler {
+  constructor() {
+    const homeDirectory = isWindows()
+      ? homedir()
+      : execCommand('eval echo ~${SUDO_USER}')
```

Comment:
if execCommand fails, GlobalCredentialsHandler will do weird things and it will be difficult to diagnose.
I suggest in a future PR:
* call here a version of execCommand that does not catch 
* In the telemetry constructor, in new GlobalCredentialsHandler(), catch it and set a flag (maybe reuse isAnalyticsEnabled) to disable the telemetry because the GlobalCredentialsHandler() is screwed

Reply:
done 👍